### ISSUE TITLE
Bug: [@typescript-eslint/no-unused-vars] import type incorrectly flagged as unused var when name of type match nested namespace name (8.11.x)

### ISSUE THREAD

**Issue by dloreto** [typescript-eslint/typescript-eslint#10300]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.ts&code=LAKALgngDgpgBAORgZzDAJnAvHVAnASwDsBzAblFCIEMBbFKagY3gBUUw4BvUOOGAB5QA9nk416yRi0QcM3Xn35DRnSLDgBBADYFqybLNQYKIRX0EixcJsKKo41QwHJnjgzr3JTfAL6h-MxBLVRs7BwAjQ3ZUADokY3RY6jIgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type Nested = string;

namespace Test {
  export namespace Nested {
    export type Alias = Nested;

    export const a = '' as Alias;
  }
}

export const b = Test.Nested.a;
```

### ESLint Config

```javascript
module.exports = {
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error",
      {
        "varsIgnorePattern": "^_",
        "argsIgnorePattern": "^_",
        "destructuredArrayIgnorePattern": "^_"
      }
    ]
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors to be reported

### Actual Result

The type used in the code is marked as unused

**Comment by bradzacher** [comment]
This is a very complicated situation to handle.
namespaces are simultaneously both values and types and also are neither values nor types -- exactly how they can be used depends on what they export.

For example
- `namespace Foo { export type T = 1 }` is a type-only namespace that can only be used in a type location.
- `namespace Bar { export const T = 1 }` is a value-only namespace that can only be used in a type location.
- `namespace Baz { export type T = 1; export const T = 1 }` is a type and value namespace that can be used in either a type or a value location.
- `namespace Wat {}` is neither a type, nor a value namespace and it cannot be used in any location.

On top of this a namespace can be declaration merged with another namespace of the same name.
And nested namespaces can be declaration merged so long as the nesting chain of names is the same.

Their exact rules are... complicated and requires type information.
So when it comes to namespaces our scope analysis is an approximation of reality.
We always assume every single namespace may be used in a type or a value location.

So when we see `type Alias = Nested` our logic roughly goes:
1) look in the scope for the closest variable named `Nested`
2) if the variable can be referenced from a type location, stop. Else goto (1)

So we see that the closed variable is the namespace and that namespace (as mentioned above) can be referenced from a type or a value location, and so it meets the criteria for us to use - so we stop.

---

There's technically something that we could do here to add complexity to the logic.
It's not technically valid to reference a namespace from a type location unless it's in a `TSQualifiedName`.
So we could feasibly try to add some logic that says "if the reference is from a type location AND the parent of the reference is NOT a `TSQualifiedName` AND the variable's `.defs` only has namespace declarations THEN it's not a valid reference".
But it's a pretty non-trivial thing to build.

cc @typescript-eslint/triage-team
My recommendation would be to close this as a very, very, very rare and expected edge case. I don't think it's worth trying to fix this given that (a) namespaces are uncommon in modern TS and (b) shadowing the name of a type with a nested namespace's name is really a 1-in-a-trillion usecase.
Thoughts?

**Comment by JoshuaKGoldberg** [comment]
> ```ts
> type Nested = string;
>
> namespace Test {
> ```

😬 

I'm also in favor of closing this. Having a `type` different from a `namespace` with the same name is spooky.

**Comment by Josh-Cena** [comment]
I'm in favor of implementing @bradzacher's suggestion, since it doesn't just pertain this single rule but pertains to TS linting at large, and if it affects other rules it may even be harder to track down.

**Comment by bradzacher** [comment]
My point was that we shouldn't bother with this edge case.
In an ideal world of infinite resources - we would favour correctness.
But here we have an edge case -- the first of its kind raised since scope analysis was introduced some 5 years ago using `namespace`s (which are rarely used in modern TS) and involves shadowing names (another practice many avoid doing).

I don't believe that it is worth us trying to derive the logic required to handle this case.

**Comment by JoshuaKGoldberg** [comment]
Closing out as out of scope. Interesting discussion though!
